kernel/mem_domain.c

```diff
@@ -198,9 +198,8 @@ void k_mem_domain_remove_partition(struct k_mem_domain *domain,
 		_arch_mem_domain_partition_remove(domain, p_idx);
 	}
 
-	domain->partitions[p_idx].start = 0;
```

Comment:
@andrewboie , please, look at this change 
It does not look necessary to clear start and attr fields, if we clear the size.

Reply:
yeah that looks fine to me